### ISSUE TITLE
ddlist is not opened 

### ISSUE THREAD

**Issue by Moo2017** [lvgl/lvgl#434]
Hi All,
I need a "Select" button and once it is pressed I need to display a list of texts. User should be able to scroll among the options (texts) and  select one option and then a confirmation box will be appeared and if user confirmed , text should be sent. To do this I created a button, a ddlist to display my list and an mbox to get confirmation. I have both hardware keypad and touch pad. I need to open the ddlist once button is pressed but once user selects the button using touch pad, list is closed while I have added lv_ddlist_open(ddlist,true). Also while object is focused , it's not selected. However, it works fine with hardware keypad. Here is my function to create ddlist ( I tried to use Roller but I came across to other issue #433 ) :

lv_obj_t * DisplayddList(lv_obj_t * parent,rep_list_t *findOptions,lv_group_t *group,uint8_t VerMargin) {

  char *optionPtr=NULL;

  if (findOptions->func!=NULL)  optionPtr=(char *)findOptions->func(findOptions->data);
  else  optionPtr=findOptions->data;

  lv_obj_t *ddlist=lv_ddlist_create(parent, NULL);
  lv_ddlist_open(ddlist,true);

  uint8_t rowNum=calcFrequencyOfChar(optionPtr,'\n');

  lv_ddlist_set_options(ddlist,optionPtr);

  if (findOptions->func!=NULL)  free(optionPtr);

  lv_obj_align(ddlist, NULL, LV_ALIGN_IN_TOP_LEFT, 10, VerMargin);

  if (rowNum>7)
  lv_ddlist_set_fix_height(ddlist,lv_obj_get_height(parent)-20);//LV_VER_RES-80);

  lv_ddlist_set_hor_fit(ddlist, false);
  lv_obj_set_width(ddlist,lv_obj_get_width(parent)-20);//

  lv_ddlist_set_action(ddlist, ddlist_action);  /*Set a function to call when anew option is chosen*/

  lv_ddlist_set_sb_mode(ddlist, LV_SB_MODE_AUTO);

  if (group!=NULL)
  {
    lv_group_add_obj(group,ddlist);
    lv_group_focus_obj(ddlist);
  }

  return ddlist;

}
Also while object is focused , it's not selected

**Comment by kisvegabor** [comment]
Hi,

`lv_ddlist_open(ddlist,true);` opens the ddlist once and not keeps it open after presser or defocus. If you something which remains open you should use a simple "lv_list".

**Comment by kisvegabor** [comment]
As this is the intended behavior of the Drop down list I close this issue.

**Comment by Moo2017** [comment]
I noticed that ddlist object is opened as soon as it is focused , is there anyway to open it only if while it is focused user presses Enter on that or touch it?

**Comment by kisvegabor** [comment]
There is no option for this now. However, it's possible improvement. If you'd like to add it I can help with the implementation.

**Comment by Moo2017** [comment]
Ok that would be great! Any help would be highly appreciated. 
I have seen another problem. While I have disabled click on the ddlist , still once I click on that it is opened.
Moreover, I'd like to know if there is any way to bring an object to foreground manually ( I know If I set lv_obj_set_top when I click on that it'll be brought on foreground ). 
Also, I don't know how I can select an option from a drop down list using touch? I created a drop down list but it always gives me the first option as a selected object.
